source/block-and-log-streamer.ts

```diff
@@ -1,40 +1,49 @@
-import { BlockHistory } from "./models/block-history";
-import { LogHistory } from "./models/log-history";
-import { reconcileBlockHistory } from "./block-reconciler";
-import { reconcileLogHistoryWithAddedBlock, reconcileLogHistoryWithRemovedBlock } from "./log-reconciler";
+import { Block } from './models/block'
+import { Log } from './models/log'
+import { Filter, FilterOptions } from './models/filters'
+import { BlockHistory } from './models/block-history'
+import { LogHistory } from './models/log-history'
+import { reconcileBlockHistory } from './block-reconciler'
+import { reconcileLogHistoryWithAddedBlock, reconcileLogHistoryWithRemovedBlock } from './log-reconciler'
 
-import { List as ImmutableList } from "immutable";
-import * as createUuid from "uuid";
+import { List as ImmutableList } from 'immutable'
+import * as createUuid from 'uuid'
 
 export interface Configuration {
 	/** number of blocks to retain in history, defaults to 100 */
 	blockRetention?: number
 }
 
 export class BlockAndLogStreamer<TBlock extends Block, TLog extends Log> {
-	private lastKnownGoodBlockHistory: BlockHistory<TBlock> = ImmutableList<TBlock>();
-	private blockHistory: Promise<BlockHistory<TBlock>> = Promise.resolve(this.lastKnownGoodBlockHistory);
-	private lastKnownGoodLogHistory: LogHistory<TLog> = ImmutableList<TLog>();
-	private logHistory: Promise<LogHistory<TLog>> = Promise.resolve(this.lastKnownGoodLogHistory);
-	private pendingCallbacks: Array<() => void> = [];
+	private lastKnownGoodBlockHistory: BlockHistory<TBlock> = ImmutableList<TBlock>()
+	private blockHistory: Promise<BlockHistory<TBlock>> = Promise.resolve(this.lastKnownGoodBlockHistory)
+	private lastKnownGoodLogHistory: LogHistory<TLog> = ImmutableList<TLog>()
+	private logHistory: Promise<LogHistory<TLog>> = Promise.resolve(this.lastKnownGoodLogHistory)
+	private pendingCallbacks: Array<() => void> = []
 
-	private readonly blockRetention: number;
+	private readonly blockRetention: number
 
-	private readonly getBlockByHash: (hash: string) => Promise<TBlock | null>;
-	private readonly getLogs: (filterOptions: FilterOptions) => Promise<TLog[]>;
-	private readonly onError: (error: Error) => void = () => {};
+	private readonly getBlockByHash: (hash: string) => Promise<TBlock | null>
+	private readonly getLogs: (filterOptions: FilterOptions) => Promise<TLog[]>
+	private readonly onError: (error: Error) => void = () => {}
 
 	private readonly logFilters: { [propName: string]: Filter } = {}
-	private readonly onBlockAddedSubscribers: { [propName: string]: (block: TBlock) => void } = {};
-	private readonly onBlockRemovedSubscribers: { [propName: string]: (block: TBlock) => void } = {};
-	private readonly onLogAddedSubscribers: { [propName: string]: (log: TLog) => void } = {};
-	private readonly onLogRemovedSubscribers: { [propName: string]: (log: TLog) => void } = {};
+	private readonly onBlockAddedSubscribers: {
+		[propName: string]: (block: TBlock) => void
+	} = {}
```

Comment:
Since this line was split but the one above it was not, I'm operating under the assumption that the `printWidth` setting in prettier is what is causing many lines in this PR to change from one to multiple.  I see the same thing in comments, function calls, conditionals etc.  Is it possible to disable prettier hard wrapping entirely?  If not, can we set it to something like `9999`?

Reply:
No, it cannot be disabled. It can be set to `99999` but this will avoid all wrapping, while you have also manually added at times. 

E.g

```
export const reconcileBlockHistory = async <TBlock extends Block>(getBlockByHash: GetBlockByHash<TBlock>, blockHistory: BlockHistory<TBlock> | Promise<BlockHistory<TBlock>>, newBlock: TBlock, onBlockAdded: (block: TBlock) => Promise<void>, onBlockRemoved: (block: TBlock) => Promise<void>, blockRetention: number = 100): Promise<BlockHistory<TBlock>> => {
```

Reply:
That is the goal, yes.  😁 

```typescript
export const reconcileBlockHistory = async <TBlock extends Block>(getBlockByHash: GetBlockByHash<TBlock>, blockHistory: BlockHistory<TBlock> | Promise<BlockHistory<TBlock>>, newBlock: TBlock, onBlockAdded: (block: TBlock) => Promise<void>, onBlockRemoved: (block: TBlock) => Promise<void>, blockRetention: number = 100): Promise<BlockHistory<TBlock>> => {
```

We can discuss if you like, but the short version of the argument is that everything after `const reconcileBlockHistory = async ...` isn't particularly valuable information for the reader.  It is super useful information for the IDE as it can provide really useful type hints, but the reader can (and likely will) glaze over it all.  By putting it all on one line we are (in essence) saying to the reader, "This is an async function named `reconcileBlockHistory`, if you want more info about the specifics of how to call this function keep reading, otherwise continue on".

Any user with an IDE will get auto-complete tooltips with the parameters formatted in a way that the UX designers for the IDE thought was most useful, and in my experience this is how people most often interact with functions in a typed language.

If this function is broken up onto multiple lines the reader needs to do additional parsing when reading past the function to mentally ignore the parameter lines.

Reply:
Ok, made the change then! Enabling word-wrap in my editor gets it the way I want it without imposing it on anyone else :)

Reply:
Looks like these specific lines still need to get rolled back.